lib/Sema/TypeCheckType.cpp

```diff
@@ -382,35 +382,35 @@ Type TypeChecker::getStringType(ASTContext &Context) {
   if (auto typeDecl = Context.getStringDecl())
     return typeDecl->getDeclaredInterfaceType();
 
-  return Type();
+  return ErrorType::get(Context);
```

Comment:
`ErrorType`'s contract is that if one has been created, a diagnostic *must* have been emitted either at the point of creation or upstream of it. We have a set of arbitrary diagnostics for when internal utilities like this fail (e.g. `diag::sugar_type_not_found`). We should emit something here and in the others.

Reply:
Ok, add a diagnostic emit for all 15 return ErrorType(ctx) in TypeCheckType.cpp and any others I added.

Reply:
For getStringType, getSubstringType, getIntType, getInt8Type, getUInt8Type, sugar_type_not_found is not really appropriate right? Since they're not really syntactic sugar for anything. It looks like these guys get called for equality checks mainly. If the underlying method comes from KNOWN_STDLIB_TYPE_DECL() in KnownStdlibTypes.def is it possible for these to fail at all? Or only if the stdlib is really broken/missing? I may have answered my question there by thinking out loud. Should I add a new diagnostic for 'broken standard library' with the associated types? or add more select cases to the existing sugar_type_not_found diagnostic?

Reply:
I started on adding to the existing sugar_type_not_found diagnostic, but emitting a diagnostic requires a source location, and these methods (and their clients) don't have one. I worked up the call chain a bit and could not find a useful source loc in the client methods (I was not exhaustive.) These getXType methods for stdlib types are kind of unusual but I don't completely understand why. I'll keep looking at it.

Reply:
I put llvm_unreachable() in the cases where there's a broken standard library. That might be OK but it would still be better to emit a diagnostic although I'm having mechanical problems emitting one in that case. 

TypeChecker::getDynamicBridgedThroughObjCClass returns ErrorType if bridging fails but TypeChecker::TypeCheckerTypedCast continues and can either return a type or generate a diagnostic. Arguably that's OK. Alternately returning ErrorType in that situation is not appropriate and there should be a UnmatchedType kind of thing to indicate that situation. 

Still need diagnostics for TypeChecker::resolveOpaqueReturnType I think. 

@CodaFi do you have suggestions or opinions about this? The above is reflected in commit 
02d34f9